packages/gatsby-theme-docs/src/components/content-notifications.js

```diff
@@ -0,0 +1,139 @@
+import PropTypes from 'prop-types';
+import { css } from '@emotion/core';
+import styled from '@emotion/styled';
+import UnstyledInformationIcon from '../icons/information-icon.svg';
+import UnstyledWarningIcon from '../icons/warning.svg';
+import UnstyledErrorIcon from '../icons/error.svg';
+import createStyledIcon from '../utils/create-styled-icon';
+import { colors, tokens, typography, dimensions } from '../design-system';
+import Spacings from './spacings';
+
+const InformationIcon = createStyledIcon(UnstyledInformationIcon);
+const WarningIcon = createStyledIcon(UnstyledWarningIcon);
+const ErrorIcon = createStyledIcon(UnstyledErrorIcon);
+
+const getIconByType = type => {
+  switch (type) {
+    case 'warning':
+      return WarningIcon;
+    case 'error':
+      return ErrorIcon;
+    default:
+      return InformationIcon;
+  }
+};
+const getIconColorByType = type => {
+  switch (type) {
+    case 'warning':
+      return 'textWarning';
+    case 'error':
+      return 'textError';
+    default:
+      return 'textInfo';
+  }
+};
+const Container = styled.div`
+  background-color: ${props => {
+    switch (props.type) {
+      case 'warning':
+        return colors.light.surfaceWarning;
+      case 'error':
+        return colors.light.surfaceError;
+      default:
+        return colors.light.surfaceInfo;
+    }
+  }};
+  border: 1px solid
+    ${props => {
+      switch (props.type) {
+        case 'warning':
+          return colors.light.textWarning;
+        case 'error':
+          return colors.light.textError;
+        default:
+          return colors.light.textInfo;
+      }
+    }};
+  border-radius: ${tokens.borderRadius6};
+  padding: ${dimensions.spacings.m};
+  min-width: 100%;
+`;
+const Flag = props => (
+  <span
+    css={css`
+      background-color: ${colors.light.surfaceInfo};
+      border: 1px solid ${colors.light.borderInfo};
+      border-radius: ${tokens.borderRadius4};
+      color: ${colors.light.textInfo};
+      font-size: ${typography.fontSizes.extraSmall};
+      padding: 0 ${dimensions.spacings.xs};
+    `}
+  >
+    {props.children}
+  </span>
+);
+Flag.propTypes = {
+  children: PropTypes.string.isRequired,
+};
+
+const ContentNotification = props => {
+  const Icon = getIconByType(props.type);
+  const iconColor = getIconColorByType(props.type);
+  return (
+    <Container type={props.type}>
+      <Spacings.Inline scale="s" alignItems="flex-start">
+        <div>
+          <Icon color={iconColor} />
+        </div>
+        {/* NOTE: we assume that the content of the elements in the MDX file
+        respects the empty lines between the component declaration, in order
+        for MDX to parse the content into React components.
+        See https://github.com/mdx-js/mdx/issues/628
+
+        Example (does not work):
+          <Info>
+            This *is* **markdown**!
+          </Info>
+
+        Example (it works):
+          <Info>
+
+            This *is* **markdown**!
+
+          </Info>
```

Comment:
/cc @nkuehn this works 👆I think we should just go with that syntax, as it's following the spec.

I haven't found another way around that without having to do the parsing on our own.

Reply:
OK, interesting.  I

I still think it's too implicit from an author's perspective, but we can opt to try it with real users and see whether they are going mad or not.  It's charming nevertheless a) because it's official and b) because an author can control in the content whether he/she wants the body to be parsed as MDX or not. 

If we find the syntax to be too implicit I don't see a problem in parsing on our own, I just opt for then re-using as much as possible from the central MDX configuration in the MDX Provider (e.g. the components).   

We do need the helper for parsing on our own in any case in the RAML generated content.  The RAML descriptions are per specification markdown and we have to render that programmatically.